PBRT Docker Support

tlian7 opened the pull request:
These changes allow users to specify a dockerFlag and run PBRT in a docker container. A dockerized version for PBRT-spectral can be found here: https://hub.docker.com/r/vistalab/pbrt/. When running the docker version, files must be copied over to a single directory.

tlian7 committed:
Remove some default values to match Scene3D output

tlian7 committed:
Add a LookAt line in the PBRT file, this seems to be missing when we use a realisticDiffraction camera.

tlian7 committed:
Large - maybe messy - commit that 1) Renders a depth map everytime you run PBRT, 2) Outputs and saves an "oi" object when rendering with PBRT, and 3) Passes parameters needed to make the oi structure to PBRT2OI (doesn't work with conditions right now)

benjamin-heasly committed:
additional mexopts.sh for newer OS X, matlab versions

tlian7 committed:
Converted back to Ben's updated RTB. Added ability to run PBRT from VistaLab's Docker container.

tlian7 committed:
Fixed pathname error for docker. Added default hints to include dockerFlag = 0.

tlian7 committed:
Fixed issues with using Mitsuba instead of PBRT

tlian7 committed:
Merge branch 'master' of https://github.com/DavidBrainard/RenderToolbox3

# By Ben Heasly
# Via Ben Heasly
* 'master' of https://github.com/DavidBrainard/RenderToolbox3:
  New example scene renders textures without mappings file.
  PBRT Collada importer supports materials with diffuse and specular textures.
  Revise Collada "clean up" to permit materials with diffuse and specular textures.
  import assets for collada texture demo
  Better reporting of compared vs skipped file pairs.
  Tolerate when compared images have different spectral depths.
  working mexopts.sh for Xcode 6.4 and Matlab 2014b
  Update Mitsuba plugin "bump" now called "bumpmap"
  All example scenes use explicit light spectra to avoid RGB promotion.
  Allow PBRT to read spectral planes up to a limit, instead of all planes.
  Explicit illuminant spectrum for point light. This is necessary because the point light was automatically imported with an RGB spectrum, and Mitsuba changed its default spectral promotion for RGB spectra.
  Utility for viewing exr-image slices.
  PBRT spectral sampling with 32 bands.
  Small typo

Conflicts:
	Admin/mexopts/XCode-6.4-Matlab-2014b/mexopts.sh
	ExampleScenes/ColladaTexture/MakeColladaTexture.m
	Utilities/ReadMultispectralEXR/ReadMultichannelEXR/MakeReadMultichannelEXR.m

tlian7 committed:
Some cleanup to help the pull request.

benjamin-heasly commented on BatchRenderer/Mappings/ResolveMappingsValues.m:
```diff
@@ -103,7 +103,19 @@
             map.right.value = fileInfo.resolvedPath;
             
             if ~fileInfo.isRootFolderMatch
```
Nice.  I like what you did here.  I think copying files to the resource folder could be useful in other cases as well.  Would you be interested in making this a separate behavior, independent of the renderer or Docker?

We could define a separate hint like `copyResources`.  Then it could be
```
if hints.copyResources
```
instead of
```
if(hints.dockerFlag == 0 || strcmp(hints.renderer,'Mitsuba'))
```

A downside would be that docker users would have to remember to set the value of this new flag.  But I think it would be worth it to gain a nice, modular behavior.

benjamin-heasly commented on RendererPlugins/PBRT/RunPBRT.m:
```diff
@@ -65,17 +65,57 @@
-renderCommand = sprintf('%s --outfile %s %s', pbrt.executable, output, sceneCopy);
-fprintf('%s\n', renderCommand);
-[status, result] = RunCommand(renderCommand, hints);
+if(hints.dockerFlag == 1)
```
Nice.  This is cool too.

May I propose 3 changes?  I think these are just fussy organizational things.
 1. How about moving the `dockerFlag` into the `pbrt` struct, rather than `hints` struct.  This would keep the Docker choice local to pbrt.  We could think of the `dockerFlag` as roughly on the same level as the pbrt executable path, which is already in the `pbrt` struct.
 2. Instead of a true/false `dockerFlag`, what about a string field like `dockerImageName`?  Then the literal value of `vistalab/pbrt` could be broken out and configured using this field instead of hard-coded into the function.  Users could leave the field empty if they want to use local pbrt.
 3. What about moving the `docker pull` command into the RTB configuration script, rather than `RunPBRT()` itself?

If you're on board with these changes, you could edit [RenderToolbox3ConfigurationTemplate](https://github.com/DavidBrainard/RenderToolbox3/blob/master/Utilities/RenderToolbox3ConfigurationTemplate.m) to add expected fields to the `pbrt` struct and put the `docker pull` command there.

Commit:
Updated SceneUtilsV1 ...

by adding camera widthToHeightAspectRatio parameter to the generateCameraType method.

benjamin-heasly committed:
allow PBRT Film to merge with adjustments

benjamin-heasly committed:
Merge branch 'master' of github.com:RenderToolbox3/RenderToolbox3

benjamin-heasly committed:
set PBRT film type from hints whether merging or not

benjamin-heasly committed:
allow Collada lookat -> PBRT LookAt

Commit:
Now rendered image in Blender has same aspect as that of the camera

tlian7 committed:
Merge branch 'master' of https://github.com/RenderToolbox3/RenderToolbox3

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
Synced with RenderToolbox3/RenderToolbox3
